DiffusionPreprocessing/scripts/DiffusionToStructural.sh

```diff
@@ -97,28 +95,57 @@ done
 ${GlobalScripts}/Rotate_bvecs.sh "$DataDirectory"/bvecs "$WorkingDirectory"/diff2str.mat "$T1wOutputDirectory"/bvecs
 cp "$DataDirectory"/bvals "$T1wOutputDirectory"/bvals
 
+DilateDistance=`echo "$DiffRes * 4" | bc`  # Extrapolates the diffusion data up to 4 voxels outside of the FOV
+
 #Register diffusion data to T1w space. Account for gradient nonlinearities if requested
 if [ ${GdcorrectionFlag} -eq 1 ]; then
     echo "Correcting Diffusion data for gradient nonlinearities and registering to structural space"
     ${FSLDIR}/bin/convertwarp --rel --relout --warp1="$DataDirectory"/warped/fullWarp --postmat="$WorkingDirectory"/diff2str.mat --ref="$T1wRestoreImage"_${DiffRes} --out="$WorkingDirectory"/grad_unwarp_diff2str
-    ${FSLDIR}/bin/applywarp --rel -i "$DataDirectory"/warped/data_warped -r "$T1wRestoreImage"_${DiffRes} -w "$WorkingDirectory"/grad_unwarp_diff2str --interp=spline -o "$T1wOutputDirectory"/data
+    ${CARET7DIR}/wb_command -volume-dilate $DataDirectory/warped/data_warped.nii.gz $DilateDistance NEAREST $DataDirectory/warped/data_warped_dilated.nii.gz
+    ${FSLDIR}/bin/applywarp --rel -i "$DataDirectory"/warped/data_warped_dilated -r "$T1wRestoreImage"_${DiffRes} -w "$WorkingDirectory"/grad_unwarp_diff2str --interp=spline -o "$T1wOutputDirectory"/data
+
+    #Create a mask covering voxels within the field of view for all volumes
+    ${FSLDIR}/bin/fslmaths "$DataDirectory"/warped/data_warped -abs -Tmax -bin "$DataDirectory"/warped/fov_mask
```

Comment:
Per my other comment, I'm thinking we should use `-bin -Tmin` instead.  (Or perhaps `-abs -bin -Tmin` if negative values are considered "legitimate").

Reply:
That's also a better match to the purpose stated in the comments.  In the general case (without knowing the internals of how `eddy` operates), `-Tmax` does not generate a mask within the field of view "for all volumes".

Reply:
I see that later in the script we remove negative intensity values from `data`, although it isn't clear to me if negative values can be present in the `eddy` output, or if they are only introduced by the spline interpolation inherent to resampling into T1w space in this script itself.

Reply:
Negative values are legitimate I think. Most importantly, if we start removing negative values this will create a lot of holes in the field of view mask and hence in the final nodif_brain_mask, which would be a problem for many analyses. To enforce the lack of hole, I actually added a -fillh. For any valid field of view mask this should not make a difference (and indeed it does not for the test subject). However, it should prevent voxels with a value that just happens to be exactly zero within the field of view from creating holes in the final mask.

Negative values are already there in the output file from eddy (before spline interpolation), but not the input file. So, I guess they arise in the internal interpolation in the eddy script.

Reply:
Why do we think negative values are legitimate?  Won't they create problems for the various model fitting we do with the diffusion data later?

Reply:
First of all, I don't think this patch should address negative values as it is unrelated with the field of view. 

Continuing this tangent anyway: For Gaussian noise models, the negative values should not be a real problem, as long as they are not too low. However, Rician noise models or the linear fitting of log(S) in dtifit do require positive values. Both of these cases would however also struggle with signal being perfectly 0, so setting negative values to 0 does not really help the diffusion modelling.

In FSL in the case of dtifit the negative values are just set to 1 (i.e., logS=0). I'm not sure how bedpostX deals with them.

Reply:
ok sounds good